src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/PinnedResolverValueResolver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\ResolverNotFoundException;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/**
+ * Delegates value resolving of arguments bearing the {@see ValueResolver} attribute to the specified resolver.
+ *
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.com>
+ */
+final class PinnedResolverValueResolver implements ValueResolverInterface
+{
+    public function __construct(private readonly ContainerInterface $resolvers)
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        if (!$attributes = $argument->getAttributesOfType(ValueResolver::class, ArgumentMetadata::IS_INSTANCEOF)) {
+            return [];
+        }
+
+        $resolverName = null;
+        foreach ($attributes as $attribute) {
+            if (!$attribute->pinned) {
```

Comment:
this pinned property doesn't make sense to me

Reply:
Related to https://github.com/symfony/symfony/pull/48992#issuecomment-1410410608